pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -279,3 +284,50 @@ func (p *criStatsProvider) makeContainerStats(
 
 	return result
 }
+
+// removeTerminatedContainer returns the specified container but with
+// the stats of the terminated containers removed.
+func removeTerminatedContainer(containers []*runtimeapi.Container, stats []*runtimeapi.ContainerStats) []*runtimeapi.Container {
+	containerMap := make(map[containerID][]*runtimeapi.Container)
+	// Sort order by create time
+	sort.Slice(containers, func(i, j int) bool {
+		return containers[i].CreatedAt < containers[j].CreatedAt
+	})
+	for _, container := range containers {
+		// Filter out the container that isn't managed by a Pod
+		if container.PodSandboxId == "" {
+			continue
+		}
+		refID := containerID{
+			podRef:        buildPodRef(container.Labels),
+			containerName: kubetypes.GetContainerName(container.Labels),
+		}
+		containerMap[refID] = append(containerMap[refID], container)
+	}
+
+	// Create container stats map.
+	containerStatsMap := make(map[string]*runtimeapi.ContainerStats)
+	for _, stat := range stats {
+		containerStatsMap[stat.Attributes.Id] = stat
+	}
+
+	result := make([]*runtimeapi.Container, 0)
+	for _, refs := range containerMap {
+		if len(refs) == 1 {
+			result = append(result, refs[0])
+			continue
+		}
+		i := 0
+		for ; i < len(refs); i++ {
+			if containerStatsMap[refs[i].Id].Cpu != nil || containerStatsMap[refs[i].Id].Memory != nil {
```

Comment:
Is it ever possible to have more than one container in refs running at once?  Why cant we just take the last element from the list?

Reply:
@dashpoleThanks for review. I have the same doubts, but cadvisor stats do like this. 
Is there any different between CRI stats and cadvisor stats? ping @yguo0905

Reply:
@dashpole Thanks for review. I have the same doubts, but cadvisor stats do like this. 
Is there any different between CRI stats and cadvisor stats? ping @yguo0905

Reply:
@dashpole Thanks for review. I have the same doubts, but cadvisor stats do like this. 
Is there any different between CRI stats and cadvisor stats? ping @yguo0905

Reply:
ping @yguo0905 
Could you please help us solve our doubts, thanks so much. :-)

Reply:
It would allow us to detect the error case where multiple containers with the same name are active at the same time. See https://github.com/kubernetes/kubernetes/issues/47853#issuecomment-310494614.